### PR TITLE
[Merged by Bors] - chore: camel `sectl` and `sectr`

### PR DESCRIPTION
**Pull request by YaelDillies**
... to `sectL` and `sectR`. I personally find these names much less confusing this way.

---

[Open in Gitpod](https://gitpod.io/from-referrer/)

### CODE REVIEW

**Review by grunweg — COMMENTED** [review]
Seems reasonable.
maintainer merge

File: Mathlib/Logic/Embedding/Basic.lean [hunk 1]
```diff
@@ -231,14 +231,18 @@ def punit {β : Sort*} (b : β) : PUnit ↪ β :=
 
 /-- Fixing an element `b : β` gives an embedding `α ↪ α × β`. -/
 @[simps]
-def sectl (α : Sort _) {β : Sort _} (b : β) : α ↪ α × β :=
+def sectL (α : Sort _) {β : Sort _} (b : β) : α ↪ α × β :=
```
Comment (b-mehta) [inline Mathlib/Logic/Embedding/Basic.lean:234]: Please add deprecated aliases for `Function.Embedding.sectl_apply` and `Function.Embedding.sectr_apply`.

File: Mathlib/CategoryTheory/Products/Basic.lean [hunk 2]
```diff
@@ -109,18 +109,22 @@ end
 -- address the universal properties.
 namespace Prod
 
-/-- `sectl C Z` is the functor `C ⥤ C × D` given by `X ↦ (X, Z)`. -/
+/-- `sectL C Z` is the functor `C ⥤ C × D` given by `X ↦ (X, Z)`. -/
 @[simps]
-def sectl (C : Type u₁) [Category.{v₁} C] {D : Type u₂} [Category.{v₂} D] (Z : D) : C ⥤ C × D where
+def sectL (C : Type u₁) [Category.{v₁} C] {D : Type u₂} [Category.{v₂} D] (Z : D) : C ⥤ C × D where
```
Comment (b-mehta) [inline Mathlib/CategoryTheory/Products/Basic.lean:114]: Please add deprecated aliases for `Prod.sectl_obj` and `Prod.sectl_map`.

File: Mathlib/CategoryTheory/Products/Basic.lean [hunk 3]
```diff
@@ -109,18 +109,22 @@ end
...
   obj X := (X, Z)
   map f := (f, 𝟙 Z)
 
-/-- `sectr Z D` is the functor `D ⥤ C × D` given by `Y ↦ (Z, Y)` . -/
+@[deprecated (since := "2024-11-12")] alias sectl := sectL
+
+/-- `sectR Z D` is the functor `D ⥤ C × D` given by `Y ↦ (Z, Y)` . -/
 @[simps]
-def sectr {C : Type u₁} [Category.{v₁} C] (Z : C) (D : Type u₂) [Category.{v₂} D] : D ⥤ C × D where
+def sectR {C : Type u₁} [Category.{v₁} C] (Z : C) (D : Type u₂) [Category.{v₂} D] : D ⥤ C × D where
```
Comment (b-mehta) [inline Mathlib/CategoryTheory/Products/Basic.lean:122]: as above

**Review by b-mehta — CHANGES_REQUESTED** [review]

File: Mathlib/CategoryTheory/Products/Basic.lean [hunk 4]
```diff
@@ -109,18 +109,25 @@ end
 -- address the universal properties.
 namespace Prod
 
-/-- `sectl C Z` is the functor `C ⥤ C × D` given by `X ↦ (X, Z)`. -/
-@[simps]
-def sectl (C : Type u₁) [Category.{v₁} C] {D : Type u₂} [Category.{v₂} D] (Z : D) : C ⥤ C × D where
+/-- `sectL C Z` is the functor `C ⥤ C × D` given by `X ↦ (X, Z)`. -/
+@[simps?]
```
Comment (b-mehta) [inline Mathlib/CategoryTheory/Products/Basic.lean:113]:
```suggestion
@[simps]
```

**Review by jcommelin — COMMENTED** [review]
Thanks :tada:

bors merge

**Pull request closed without merging**